models/domains_tlds.php

```diff
@@ -140,12 +140,12 @@ public function add(array $vars)
 
                 // If a module id is not provided, use the None module by default
                 if (!isset($vars['module_id'])) {
-                    if (!$this->ModuleManager->isInstalled('none', $vars['company_id'])) {
-                        $this->ModuleManager->add(['class' => 'none', 'company_id' => $vars['company_id']]);
+                    if (!$this->ModuleManager->isInstalled('generic_domain', $vars['company_id'])) {
```

Comment:
Looking at this again it really feels like an unnecessary special case that isn't really reliable.

I'm thinking we should remove lines 141-154 here and update lines 137-139 to remove the condition as it will no longer be necessary.
We should also change the validation rule for module_id to 'if_set' => $edit.

Reply:
Updated